fw/rbcx-coprocessor/src/SmartServoController.cpp

```diff
@@ -0,0 +1,187 @@
+#include "SmartServoController.hpp"
+#include "Bsp.hpp"
+#include "ControlLink.hpp"
+#include "stm32f1xx_ll_gpio.h"
+#include "stm32f1xx_ll_usart.h"
+#include "utils/Debug.hpp"
+#include "utils/TickTimer.hpp"
+
+enum SmartServoState { IDLE, TX, RX };
+
+static SmartServoState gState = IDLE;
+static std::array<uint8_t, 16> gBuff;
+static size_t gBuffSize = 0;
+static size_t gBuffIndex = 0;
+static bool gExpectResponse = false;
+
+static CoprocStat_SmartServoStat gResponse;
+static bool gResponseReady = false;
+static TickTimer gResponseTimeout;
+
+static TickTimer gStateTimeout;
+
+void smartServoInit() {
+    LL_USART_InitTypeDef init;
+    LL_USART_StructInit(&init);
+    init.BaudRate = 115200;
+    init.DataWidth = LL_USART_DATAWIDTH_8B;
+    init.HardwareFlowControl = LL_USART_HWCONTROL_NONE;
+    init.Parity = LL_USART_PARITY_NONE;
+    init.StopBits = LL_USART_STOPBITS_1;
+    init.TransferDirection = LL_USART_DIRECTION_TX;
+
+    if (LL_USART_Init(servoUart, &init) != SUCCESS)
+        abort();
+
+    LL_USART_ConfigHalfDuplexMode(servoUart);
+    LL_USART_Enable(servoUart);
+
+    HAL_NVIC_SetPriority(servoUartIRQn, servoUartIrqPrio, 0);
+    HAL_NVIC_EnableIRQ(servoUartIRQn);
+
+    LL_USART_DisableIT_ERROR(servoUart);
+    LL_USART_DisableIT_IDLE(servoUart);
```

Comment:
Hmm tyhle bity by neměly být nastavený po resetu.. ale to je asi jedno

Reply:
Taky mi přišlo, ale skákalo mi to do interruptu s error flagama když jsem odpojoval/zapojoval servo - ještě to otestuju.